components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -775,18 +776,43 @@ IJsonSerializer jsonSerializer
         }
 
         /// <inheritdoc/>
-        public async Task<PublishedNodesEntryModel> GetConfiguredNodesOnEndpointAsync(
+        public async Task<List<OpcNodeOnEndpointModel>> GetConfiguredNodesOnEndpointAsync(
             PublishedNodesEntryModel request,
             CancellationToken ct = default) {
 
             _logger.Information("{nameof} method triggered", nameof(GetConfiguredNodesOnEndpointAsync));
+            var sw = Stopwatch.StartNew();
             await _lockConfig.WaitAsync(ct).ConfigureAwait(false);
+            List<OpcNodeOnEndpointModel> response = new List<OpcNodeOnEndpointModel>();
```

Comment:
Please switch lines `785` and `786` so that acquiring the lock is immediately followed by the `try/catch/finally` block.

Reply:
switched